Sources/ComposableArchitecture/TestStore.swift

```diff
@@ -2316,6 +2316,20 @@ class TestReducer<State, Action>: ReducerProtocol {
         )
         .map { .init(origin: .receive($0), file: action.file, line: action.line) }
         .eraseToEffect()
+
+    case .run(let priority, let perform):
+      let effect = LongLivingEffect(action: action)
+      return EffectTask.run(priority: priority) { [effectDidSubscribe, weak self] send in
+        self?.inFlightEffects.insert(effect)
+        Task {
+          await Task.megaYield()
+          effectDidSubscribe.continuation.yield()
+        }
+        await perform(send)
+        // FIXME: is this thread-safe?
+        self?.inFlightEffects.remove(effect)
```

Comment:
@mbrandonw is `TestReducer` intended to be `MainActor`-isolated? If so, should this `run` task be performed on the MainActor too? Given that `self` isn't Sendable I think this line (and the `insert` above) could introduce a race condition unless we lock/dispatch to the main thread.